snuba/datasets/entities/discover.py

```diff
@@ -459,6 +460,7 @@ def callback_func(results: List[Result[QueryResult]]) -> None:
                             logger.warning(
                                 "Non matching Discover result - different result",
                                 extra={
+                                    "query": query.get_body(),
```

Comment:
This will be empty with SnQL. We should not rely on it.
More important, which kinds of discrepancies are we observing as of now? Are we talking of queries that give a completely different result that could make us think there is an actual issue in how one of the two storages (errors/events) is processed, or are we talking of queries that may just return different results because the data is not perfectly aligned (thus where a similarity score would help).

Reply:
As of now, I would assume that majority of the discrepancies are related to 1) the errors storage not being backfilled yet, and 2) the events and errors consumers (and replacers) not being in sync. These two sources of differences should drop off soon. But having visibility on what the actual queries are will help diagnose this better.

Reply:
ok, just bear in mind the query may be empty

Reply:
I switched it to query.get_all_expressions(), but I'm not sure if this string will be too long/contain all relevant details - hence would prefer to revert for now